src/filesystem_helper.cpp

```diff
@@ -306,6 +306,22 @@ path temp_directory_path()
   return path(temp_path);
 }
 
+path create_temp_directory(const std::string & base_name, const path & parent)
+{
+  auto template_path = base_name + ".XXXXXX";
+
+#ifdef _WIN32
+  _mktemp_s(&template_path[0], template_path.size() + 1);
```

Comment:
@emersonknapp also, why isn't `_mktemp_s` given a full path? How will it check for file existence?